tests/lava/proc/conv/test_models.py

```diff
@@ -15,23 +17,32 @@
 
 verbose = True if (('-v' in sys.argv) or ('--verbose' in sys.argv)) else False
 TORCH_IS_AVAILABLE = utils.TORCH_IS_AVAILABLE
+seed = np.random.randint(1000)
+np.random.seed(seed)
```

Comment:
Maybe I am missing something but setting the random seed to a randomly generated number does not make unit tests predictable. I think Andreas as suggesting to set the seed to a fixed value, or to run the test in a loop, where each iteration sets the random seed to a (different) fixed value.

Everything else looks great.

Reply:
Indeed ;-)

Reply:
What I want in the unittest is to have unpredictability. That is the point of having the argument dimensions randomly generated. If we set predictable 10 seed values, then the test would only test for that combination of 10 settings. What if there is issue in any case that is not covered in the 10 settings? I am trying to do this because we would not want to slow down the build.
With unpredictablity, every time unittest is run, we are testing different setting and the testing confidence grows over time. Once such failure seeds are found, those would then be used to add more custom test for that specific setting. I think this is called monkey testing.
The logic above is just keeping track of seed so that was used in the experiment which would show up once the test fails and we can use the seed to replicate the exact setting for with the test failed and investigate more.

Reply:
Perhaps we can use fixed seeds here and design an elaborate test for nightly/weekly test?

Reply:
Okay, I googled around for and against random testing. It's not a good idea to randomize CI/CD test esp for large projects. My bad. I'll switch to fixed seeds.

Reply:
No worries, Sumit. I completely understand your reasoning and I would often like to do the same thing.

We could think about setting up a separate infrastructure for large-scale tests that need time to run, for instance for complete grid searches. These tests would then be decoupled from the regular CI/CD and would only be run from time to time on our servers. Something like that.
Maybe there is an established best practice for that @mgkwill ?